smartapps/smartthings/lifx-connect.src/lifx-connect.groovy

```diff
@@ -375,6 +375,14 @@ def devicesList(selector = '') {
         } else if (resp.status == 401) {
             log.warn "Access token is not valid"
             state.lifxAccessToken = null
+        } else if (resp.status == 404 && resp.data.contains('Could not find location_id')) {
+            log.warn "Location is not valid"
+            devices = devicesList()
```

Comment:
Is there any possibility of a recursive loop here since we are calling `devicesList()` from inside `devicesList()`? Maybe we should check explicitly on the selector not being empty before making the recursive call. As it is we are depending on the response to prevent the recursive loop.